bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/CallOptionsFactory.java

```diff
@@ -71,17 +77,28 @@ public ConfiguredCallOptionsFactory(CallOptionsConfig config) {
     }
 
     @Override
-    public <RequestT> CallOptions create(
-        MethodDescriptor<RequestT, ?> descriptor, RequestT request) {
-      if (!config.isUseTimeout() || request == null) {
+    /**
+     * Creates a {@link CallOptions} with a focus on {@link Deadlines}.  Deadlines are decided in the following order:
+     * <ol>
+     *     <li> If a user set a  {@link Context} deadline (see {@link Context#getDeadline()}), use that</li>
+     *     <li> If a user configured deadlines via {@link CallOptionsConfig}, use it.</li>
+     *     <li> Otherwise, use {@link CallOptions#DEFAULT}.</li>
+     * </ol>
+     *
+     */
+    public <RequestT> CallOptions create(MethodDescriptor<RequestT, ?> descriptor, RequestT request) {
+      Deadline contextDeadline = Context.current().getDeadline();
+      if (contextDeadline != null) {
+        return CallOptions.DEFAULT.withDeadline(contextDeadline);
+      } else if (!config.isUseTimeout() || request == null) {
```

Comment:
nit: Can we have the same order as in the comment?
- use deadline from context
- use deadline from config if provided
- use DEFAULT